lib/shape/data_visitor.rb

```diff
@@ -8,13 +8,15 @@ def visit(visitor = lambda {|x| x})
   protected
 
     def properties_to_visit
+      all_properties.select { |_, property| visitable?(property) }
+    end
+
+    def all_properties
       self.class.properties.merge(self.properties)
     end
 
     def data_visitor(properties = self.properties_to_visit, visitor = lambda {|x| x})
       properties.each_with_object({}) do |(name, property), obj|
-        next unless visitable?(property)
```

Comment:
Hmm, this means that when passing in properties manually, the filter won't be applied